cachi2/interface/cli.py

```diff
@@ -95,11 +98,22 @@ def _looks_like_json(value: str) -> bool:
     return value.lstrip().startswith(("{", "["))
 
 
+class ArgsFile(BaseModel, extra="forbid"):
+    packages: list[dict[str, str]] = []
+    flags: list[str] = []
```

Comment:
Maybe this should be an input model?

```python
class SomeArgs(BaseModel, extra="forbid"):
    """Args that you can set in the args file."""
    packages: list[PackageInput]
    flags: list[Flag]


class Request(PackageArgs):
     """All args for a fetch-deps invocation."""
    source_dir: Path
    ...
```

Reply:
I initially wrote it there... but I think this is very tied to the CLI itself, so I moved it to here at least for now. In the end, this will need to become a valid Request with InputPackages, which sound more generic.

Reply:
Hmm, it feels strange to have a "valid" ArgsFile with invalid packages and flags

Reply:
Yeah, but that would be caught up in the input model validation, whereas an empty file + stdin packages would not.

Reply:
How do we want to proceed with this? I still think the args file is more CLI related than anything else, and the use of Pydantic model here was to:
- fail on extra parameters in the file
- fail in case packages was missing or empty
- the code looks nicer in a class

I'd argue that having a valid ArgsFile with invalid packages is the same as passing the CLI args validation for having a valid JSON, but having it fail in the actual input model validation.

I can move it if you feel like it belongs to the input package, and add real `InputPackages`/`Flags` to the class.